idioms/README.md

```diff
@@ -1,3 +1,9 @@
 # Idioms
 
-TODO: add description/explanation
+[Idioms](https://en.wikipedia.org/wiki/Programming_idiom) are commonly used styles and patterns largely agreed upon by a community. They are guidelines. Writing idiomatic code allows other developers to understand what is happening because they are familiar with the form that it has.
+
+The computer understands the machine code that is generated by the compiler. The language is therefore mostly beneficial to the developer. So, since we have this abstraction layer, why not put it to good use and make it simple?
+
+Remember the [KISS principle](https://en.wikipedia.org/wiki/KISS_principle): "Keep It Simple, Stupid". It claims that "most systems work best if they are kept simple rather than made complicated; therefore, simplicity should be a key goal in design, and unnecessary complexity should be avoided".
```

Comment:
This sounds like go to me. I feel like rust is aiming to be "keep it simple but not simpler". Go is able to be simple by having tons of footguns as a result. Rust I would say aims more to be correct than simple (this is why rust became so complicated), one good thing is that the footguns are reduced to bare minimum.

Reply:
I strongly agree with you.

We can say something like:
"even if the kiss principle is important rust tends to prefer correctness over simplicity."

If you want to do a PR where you explain this it would be nice I think (you don't have to report my words necessarily).